src/shared/components/common/emoji-picker.tsx

```diff
@@ -0,0 +1,99 @@
+import { Component } from "inferno";
+import { i18n } from "../../i18next";
+import { getEmojiMart } from "../../utils";
+import { Icon } from "./icon";
+
+
+interface EmojiMartProps {
+    onEmojiClick?(val: any): any;
+    pickerOptions: any;
+}
+
+interface EmojiMartState {
+}
+
+export class EmojiMart extends Component<
+    EmojiMartProps,
+    EmojiMartState
+> {
+    private emptyState: EmojiMartState = {
+    };
+    constructor(props: any, context: any) {
+        super(props, context);
+        this.state = this.emptyState;
+    }
+    componentDidMount() {
+        let div: any = document.getElementById("emoji-picker");
+        if (div) {
+            div.appendChild(getEmojiMart(this.handleEmojiClick,this.props.pickerOptions));
+        }
+    }
+
+    render() {
+        return (
+            <div id="emoji-picker"></div>
+        );
+    }
+
+    handleEmojiClick = (e: any) => {
+        if (this.props.onEmojiClick)
+            this.props.onEmojiClick(e);
+    }
+}
+
+interface EmojiPickerProps {
+    onEmojiClick?(val: any): any;
+}
+
+interface EmojiPickerState {
+    showPicker: boolean,
+}
+
+export class EmojiPicker extends Component<
+    EmojiPickerProps,
+    EmojiPickerState
+> {
+    private emptyState: EmojiPickerState = {
+        showPicker: false,
+    };
+    constructor(props: any, context: any) {
+        super(props, context);
+        this.state = this.emptyState;
+    }
+    render() {
+        return (
+            <span>
+                <button
+                    className="btn btn-sm text-muted"
+                    data-tippy-content={i18n.t("emoji")}
+                    aria-label={i18n.t("emoji")}
+                    onClick={this.togglePicker}
+                >
+                    <Icon icon="smile" classes="icon-inline" />
+                </button>
+
+                {this.state!.showPicker && (
+                    <>
+                        <div className="emoji-picker-container">
+                            <EmojiMart onEmojiClick={this.handleEmojiClick} pickerOptions={({})}></EmojiMart>
+                        </div>
+                        <div
+                            onClick={this.togglePicker}
+                            className="click-away-container"
+                        />
+                    </>
+                )}
+            </span>
+        );
+    }
+
+    togglePicker = (e) => {
```

Comment:
Functions, not closures.

Reply:
I fixed this, but curious why? since it avoids the need to bind(this) on the function in constructor.